source/posts/2020-09-10-cross-browser-testing-with-non-functional-tests.html.markdown.erb

```diff
@@ -0,0 +1,194 @@
+relying on functional tests. Not all cross browser compatibility issues are 
+caught by functional tests. Here, I want to highlight how teams are using 
+alternatives to address them.
+
+## Deal with Non Functional Requirements earlier
+
+From the definition of BDD or Behaviour Driven Development, we know that,
+
+> The plan for implementing non-functional requirements is detailed in 
+the [system architecture](https://en.wikipedia.org/wiki/Systems_architecture), 
+because they are usually [architecturally significant requirements](https://en.wikipedia.org/wiki/Architecturally_significant_requirements). 
+- [wikipedia](https://en.wikipedia.org/wiki/Non-functional_requirement).
+
+Cross browser requirements have to be taken into account by developers 
+before the development begins as it’s architecturally significant.
+
+Most probably everyone uses some kind of framework which already checks 
+for compatibility. Javascript frameworks like [ReactJS](https://reactjs.org/), 
+[vue.js](https://vuejs.org/), [jquery](https://jquery.com/), and UI frameworks 
+like [bootstrap](https://getbootstrap.com/), [tailwind](https://tailwindcss.com/) 
+test for cross browser compatibility.
+
+Here are a few blogs that talk about the same
+
+* “React implements a totally different DOM system which is independent 
+of the browser. The implementation is done to make the application 
+cross-browser compatible.” - Cross-Browser Compatibility and React.js Web Apps
+* “Vue.js guide to ensure cross browser compatibility” - Browser Compatibility
+* “If you are using Twitter Bootstrap, and have been curious enough to 
+inspect the core bootstrap.css stylesheet, you will find approximately 
+6,000 lines of pure CSS code.  If you delve in deeper and look at the style 
+code in detail, you will find how they use multiple tricks and techniques to 
+make sure Bootstrap is the most cross browser CSS framework” 
+- [cross browser compatibility tips](https://www.monitis.com/blog/cross-browser-compatibility-tips-part-1/)
+
+It is better to invest in the right architecture for your application than 
+waiting for issues to be reported.
+
+## Write non functional tests for Cross browser compatibility
+
+To understand what we expect from a cross browser test, let’s analyse how 
+the browser is rendered. There are 3 main components to how browser 
+rendering works. 
+
+* HTML - handles the static structure of the page
+* CSS - is the way we present the information with styles and visual layout 
+* Scripts - are used to give behaviour to the UI elements.
+
+{:.specialParagraphTag}
+The functionality of any software does not change with the browser in which it 
+is executing. To ensure that the software functions as expected in the browser/s 
+one has to ensure that the CSS,HTML and the features of the scripting language 
+used works in the respective browsers. 
+This can be achieved using static analysis of front end code and does not need 
+dynamic analysis.
+
```

Comment:
Change the whole para to

This can be done by static analysis of front end code without running the application.